projects/aca-content/folder-rules/src/rule-details/conditions/rule-simple-condition.ui-component.ts

```diff
@@ -147,6 +192,63 @@ export class RuleSimpleConditionUiComponent implements ControlValueAccessor, OnD
   }
 
   ngOnDestroy() {
-    this.formSubscription.unsubscribe();
+    this.onDestroy$.next();
+    this.onDestroy$.complete();
+  }
+
+  ngOnInit() {
+    this.form.valueChanges.pipe(takeUntil(this.onDestroy$)).subscribe((value: any) => {
+      this.onChange(value);
+      this.onTouch();
+    });
+
+    this.fieldControl.valueChanges.pipe(distinctUntilChanged(), takeUntil(this.onDestroy$)).subscribe((field: string) => {
+      if (field === 'category') {
+        this.autoCompleteOptionsSubscription = this.form
+          .get('parameter')
+          .valueChanges.pipe(distinctUntilChanged(), debounceTime(AUTOCOMPLETE_OPTIONS_DEBOUNCE_TIME), takeUntil(this.onDestroy$))
+          .subscribe((categoryName) => {
+            this.getCategories(categoryName);
+          });
+        this.parameterControl.setValue('');
+      } else {
+        this.autoCompleteOptionsSubscription?.unsubscribe();
+      }
+    });
+  }
+
+  private getCategories(categoryName: string) {
+    this.showLoadingSpinner = true;
+    this.categoryService
+      .searchCategories(categoryName)
+      .pipe(first())
+      .subscribe((existingCategoriesResult) => {
+        this.showLoadingSpinner = false;
+        const options: AutoCompleteOption[] = existingCategoriesResult?.list?.entries?.map((rowEntry) => {
+          const option = new AutoCompleteOption();
```

Comment:
Looks similar to the code in line 155, maybe we can move autocomplete option mapping to a separate helper function?

Reply:
I thought of making a single helper function as well, but in the 2 codes, there is some slight difference in the object we are dealing with. One has a type of CategoryEntry, and the other has a type of ResultSetRowEntry. This makes a difference in how we fetch the complete path for the display label. 

For Category entry, we do 
```
category.entry.path.split('/').splice(3).join('/')
```

For ResultSetRowEntry, we do 
```
rowEntry.entry.path.name.split('/').splice(3).join('/')
```

I've made a helper function with the 2 types shown below. If this looks fine, then I can push the code and we can go ahead with it
```
  buildAutoCompleteOptionFromCategory(category: CategoryEntry | ResultSetRowEntry): AutoCompleteOption {
    let path = '';
    if (category instanceof CategoryEntry) {
      path = category.entry.path.split('/').splice(3).join('/');
    } else if (category instanceof ResultSetRowEntry) {
      path = category.entry.path.name.split('/').splice(3).join('/');
    }
    const option: AutoCompleteOption = {
      value: category.entry.id,
      displayLabel: path ? `${path}/${category.entry.name}` : category.entry.name
    };
    return option;
  }
```

Reply:
I think that you could also build this function in a bit different way. Instead of passing the whole objects you could just pass `id`, `path` and `name`. Then you could use the same logic to build an option for both objects and the difference would just be in the function calls in the right places

Reply:
Updated the buildAutoCompleteOption method. It now takes in 3 arguments, categoryId, categoryPath, and categoryName, and generates an autocomplete option with the correct path accordingly